contracts/src/Cids.sol

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+library Cids {
+    // TODO PERF: https://github.com/FILCAT/pdp/issues/16#issuecomment-2329836995 
+    struct Cid {
+        bytes data;
+    }
+
+    function digestFromCid(Cid memory cid) public pure returns (bytes32) {
+        require(cid.data.length >= 32, "Cid data is too short");
+        bytes memory dataSlice = new bytes(32);
+        // This can probably be optimised in assembly.
```

Comment:
Its probably worth doing that here right now.  I remember [playing around with mload](https://github.com/FILCAT/client-contract/blob/zen/dev/src/CBORParse.sol#L95) to do something similar a long time ago and it was pretty simple.